pkg/nrtupdater/node_getter.go

```diff
@@ -0,0 +1,68 @@
+package nrtupdater
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type NotConfigured struct{}
+
+func (err NotConfigured) Error() string {
+	return "unconfigured feature"
+}
+
+type ConnectionError struct {
+	Err error
+}
+
+func (err ConnectionError) Error() string {
+	return fmt.Sprintf("error connection k8s: %v", err.Err)
+}
+func (err ConnectionError) Unwrap() error {
+	return err.Err
+}
+
+func IsNotConfiguredNodeGetter(err error) bool {
+	_, ok := err.(NotConfigured)
+	return ok
+}
+
+type NodeGetter interface {
+	Get(ctx context.Context, nodeName string, opts metav1.GetOptions) (*corev1.Node, error)
+}
+
+type DisabledNodeGetter struct {
+}
+
+func (ng *DisabledNodeGetter) Get(ctx context.Context, nodeName string, opts metav1.GetOptions) (*corev1.Node, error) {
+	return nil, NotConfigured{}
+}
+
+type OneshotNodeGetter struct {
+	K8SCli kubernetes.Interface
+	node   *corev1.Node
+}
+
+func (ng *OneshotNodeGetter) Get(ctx context.Context, nodeName string, opts metav1.GetOptions) (*corev1.Node, error) {
+	if ng.node != nil {
+		return ng.node, nil
+	}
+
+	node, err := ng.K8SCli.CoreV1().Nodes().Get(ctx, nodeName, opts)
+	if err != nil {
+		return nil, ConnectionError{Err: err}
+	}
+	ng.node = node.DeepCopy()
+	return ng.node, nil
+}
+
+func NewNodeGetter(enabled bool, k8xcli kubernetes.Interface) NodeGetter {
+	if enabled {
+		return &OneshotNodeGetter{K8SCli: k8xcli}
+	}
+	return &DisabledNodeGetter{}
+}
```

Comment:
thanks, this is the right direction. Even better would be to have an object which wraps a map nodeName -> nodeObject, initialize it with the output of `List` (once at startup) and return the cached data with `Get`.
Nodes should come and go rarely, and if updates are needed we can just restart the RTE.
Once this is in place, a followup could be a GC process (compare the node listing with the NRT listing) like NFD implemented lately. Stuff for a followup PR.